File: src/luogu/p5716/Main.cpp
// https://www.luogu.com.cn/problem/P5716
#include <cstdio>

int main() {
  int y, m;
  scanf("%i%i", &y, &m);
  int arr[13] = {0, 31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
  if (m == 2 && ((y % 4 == 0 && y % 100 != 0) || y % 400 == 0)) {
    printf("29\n");
  } else {
    printf("%i\n", arr[m]);
  }
  return 0;
}

File: src/luogu/p5716/Main.py
# https://www.luogu.com.cn/problem/P5716
s = input().split()
y = int(s[0])
m = int(s[1])
arr = (0, 31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31)
if m == 2 and ((y % 4 == 0 and y % 100 != 0) or y % 400 == 0):
    print(29)
else:
    print(arr[m])
